entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -27,75 +28,95 @@
             .build(
-                new CacheLoader<>() {
-                  public Entity load(@Nonnull EdsCacheKey key) throws Exception {
-                    Entity entity = client.getById(key.tenantId, key.entityId);
-                    if (entity == null) {
-                      throw new NotFoundException("Entity not found");
-                    }
-                    return entity;
-                  }
-                });
+                CacheLoader.asyncReloading(
+                    new CacheLoader<>() {
+                      public Entity load(@Nonnull EdsCacheKey key) throws Exception {
+                        Entity entity = client.getById(key.tenantId, key.entityId);
+                        if (entity == null) {
+                          throw new NotFoundException("Entity not found");
+                        }
+                        return entity;
+                      }
+                    },
+                    executor));
 
     this.entityIdsCache =
         CacheBuilder.newBuilder()
+            .refreshAfterWrite(cacheConfig.getEntityIdsCacheRefreshMs(), TimeUnit.MILLISECONDS)
             .expireAfterWrite(cacheConfig.getEntityIdsCacheExpiryMs(), TimeUnit.MILLISECONDS)
             .maximumSize(cacheConfig.getEntityIdsMaxCacheSize())
             .recordStats()
             .build(
-                new CacheLoader<>() {
-                  public String load(@Nonnull EdsTypeAndIdAttributesCacheKey key) throws Exception {
-                    Entity entity =
-                        client.getByTypeAndIdentifyingAttributes(
-                            key.tenantId, key.byTypeAndIdentifyingAttributes);
-                    if (entity == null) {
-                      throw new NotFoundException("Entity not found!!");
-                    }
-                    entityCache.put(
-                        new EdsCacheKey(entity.getTenantId(), entity.getEntityId()), entity);
-                    return entity.getEntityId();
-                  }
-                });
+                CacheLoader.asyncReloading(
+                    new CacheLoader<>() {
+                      public String load(@Nonnull EdsTypeAndIdAttributesCacheKey key)
+                          throws Exception {
+                        Entity entity =
+                            client.getByTypeAndIdentifyingAttributes(
+                                key.tenantId, key.byTypeAndIdentifyingAttributes);
+                        if (entity == null) {
+                          throw new NotFoundException("Entity not found!!");
+                        }
+                        entityCache.put(
+                            new EdsCacheKey(entity.getTenantId(), entity.getEntityId()), entity);
+                        return entity.getEntityId();
+                      }
+                    },
+                    executor));
+    PlatformMetricsRegistry.registerCache(
+        this.getClass().getName() + ".enrichedEntityCache",
```

Comment:
Curious: Any reason for prefixing with the class name?

Reply:
easier to track which class holds the caches and also avoid naming conflicts